api/agent/agent.go

```diff
@@ -752,8 +761,10 @@ func (a *agent) prepCold(ctx context.Context, call *call, tok ResourceToken, ch
 
 	call.containerState.UpdateState(ctx, ContainerStateStart, call.slots)
 
+	deadline := time.Now().Add(time.Duration(call.Timeout) * time.Second)
+
 	// add Fn-specific information to the config to shove everything into env vars for cold
-	call.Config["FN_DEADLINE"] = strfmt.DateTime(call.execDeadline).String()
+	call.Config["FN_DEADLINE"] = strfmt.DateTime(deadline).String()
```

Comment:
this value won't match up very well, e.g. if it gets hung up waiting on a resource token. think for hot this gets built at call create time, can reuse that value. also, i'm not sure anybody is really using this, though the fdk-go is so anyone using that

Reply:
So, the new timeout semantics (FN_DEADLINE) start at slot.exec() time for hot, although for cold it's not super accurate, it's close...

Reply:
right, the other way would be the issue. pull would throw this off a good bit?

Reply:
We could split this and move pull/prep into prepCold and rest to slot.exec(), but cold is going away...